src/lazy/text/buffer.rs

```diff
@@ -3046,6 +3057,22 @@ mod tests {
         ]
     }
 
+    matcher_tests_with_macro! {
+        allow_omitting_trailing_optionals
+        match_e_expression
+        "(macro foo (a b+ c? d*) null)"
+        expect_match: [
```

Comment:
Worth adding an expectation that the explicit empty argument group `(:)` also works?